sdk/identity/src/token_credentials/auto_refreshing_credentials.rs

```diff
@@ -10,10 +11,11 @@ fn is_expired(token: &TokenResponse) -> bool {
 }
 
 #[derive(Clone)]
-/// Wraps a TokenCredential and handles token refresh on token expiry
+/// Wraps a TokenCredential and handles token refresh on token expiry.
 pub struct AutoRefreshingTokenCredential {
     credential: Arc<dyn TokenCredential>,
-    current_token: Arc<RwLock<Option<azure_core::Result<TokenResponse>>>>,
+    // Tokens are specific to a resource, so we cache tokens by resource.
+    token_cache: Arc<RwLock<HashMap<String, azure_core::Result<TokenResponse>>>>,
```

Comment:
I recognize this is existing behavior, but could we move to only caching the `TokenResponse`, not the `Result<TokenResponse>`?   We only use the value if it's `Ok`.

Reply:
Yep - excellent idea.  Done.